Translation Unit Testing

wendyadi opened the pull request:
Translation Halaman unit testing untuk issue #31

wendyadi committed:
Translation Unit Testing

Translation Halaman unit testing untuk issue #31

wendyadi commented:
@nusendra 
Mohon feedbacknya
Ada beberapa istilah yang saya bingung apakah baiknya di translate atau tidak?

Contohnya:
assertion, options, raw options, component, testable

mazipan commented on src/v2/guide/unit-testing.md:
```diff
@@ -4,11 +4,12 @@ type: guide
 ---
 
-> [Vue CLI](https://cli.vuejs.org/) has built-in options for unit testing with [Jest](https://github.com/facebook/jest) or [Mocha](https://mochajs.org/) that works out of the box. We also have the official [Vue Test Utils](https://vue-test-utils.vuejs.org/) which provides more detailed guidance for custom setups.
+> [Vue CLI](https://cli.vuejs.org/) memiliki opsi built-in untuk melakukan unit testing menggunakan [Jest](https://github.com/facebook/jest) atau [Mocha](https://mochajs.org/) yang bisa berjalan tanpa setup tambahan. Kami juga memiliki library resmi [Vue Test Utils](https://vue-test-utils.vuejs.org/) yang menyediakan tuntunan lebih detail untuk custom setups.
```
built-in -> bawaan

mazipan replied:
library -> pustaka

mazipan replied:
Setup kayanya lebih baik diterjemahkan jadi penyetelan atau setelan

mazipan commented on src/v2/guide/unit-testing.md:
```diff
@@ -4,11 +4,12 @@ type: guide
-## Simple Assertions
 
-You don't have to do anything special in your components to make them testable. Export the raw options:
+## Pengujian/Assertion Sederhana
```
Pengujian (*Assertion*) Sederhana

mazipan commented on src/v2/guide/unit-testing.md:
```diff
@@ -4,11 +4,12 @@ type: guide
-You don't have to do anything special in your components to make them testable. Export the raw options:
+## Pengujian/Assertion Sederhana
+
+Anda tidak perlu melakukan hal khusus untuk membuat komponen-komponen Anda testable (dapat dites). Export raw options:
```
bisa dites (*testable*)

mazipan replied:
Cukup ekspor opsi seperti berikut:

nusendra commented:
@wendyadi untuk istilah yang kalau dijadikan indonesia malah tambah membingungkan, tetapi saja pakai versi inggris nya, cuman dibikin italic saja

untuk component diganti jadi komponen

mazipan commented on src/v2/guide/unit-testing.md:
````diff
@@ -29,36 +30,38 @@ You don't have to do anything special in your components to make them testable.
 ```
 
-Then import the component options along with Vue, and you can make many common assertions (here we are using Jasmine/Jest style `expect` assertions just as an example):
+Kemudian import options komponen tersebut bersama dengan Vue, dan Anda dapat membuat banyak assertion umum (disini kita menggunakan Jasmine/Jest style `expect` assertion sebagai contoh)
````
Kemudian impor opsi komponen

mazipan replied:
Kata asing kaya assertion mesti pakai italic mas

mazipan replied:
Jasmine/Jest style -> gaya Jasmine/Jest

nusendra commented on src/v2/guide/unit-testing.md:
```diff
@@ -4,11 +4,12 @@ type: guide
 ---
 
-> [Vue CLI](https://cli.vuejs.org/) has built-in options for unit testing with [Jest](https://github.com/facebook/jest) or [Mocha](https://mochajs.org/) that works out of the box. We also have the official [Vue Test Utils](https://vue-test-utils.vuejs.org/) which provides more detailed guidance for custom setups.
+> [Vue CLI](https://cli.vuejs.org/) memiliki opsi built-in untuk melakukan unit testing menggunakan [Jest](https://github.com/facebook/jest) atau [Mocha](https://mochajs.org/) yang bisa berjalan tanpa setup tambahan. Kami juga memiliki library resmi [Vue Test Utils](https://vue-test-utils.vuejs.org/) yang menyediakan tuntunan lebih detail untuk custom setups.
```
```suggestion
> [Vue CLI](https://cli.vuejs.org/) memiliki opsi *built-in* untuk melakukan *unit testing* menggunakan [Jest](https://github.com/facebook/jest) atau [Mocha](https://mochajs.org/) yang bisa berjalan tanpa konfigurasi tambahan. Kami juga memiliki pustaka resmi [Vue Test Utils](https://vue-test-utils.vuejs.org/) yang menyediakan panduan lebih detail untuk konfigurasi kustom.
```

nusendra commented on src/v2/guide/unit-testing.md:
```diff
@@ -4,11 +4,12 @@ type: guide
-You don't have to do anything special in your components to make them testable. Export the raw options:
+## Pengujian/Assertion Sederhana
+
+Anda tidak perlu melakukan hal khusus untuk membuat komponen-komponen Anda testable (dapat dites). Export raw options:
```
```suggestion
Anda tidak perlu melakukan hal khusus untuk membuat komponen-komponen Anda *testable* (dapat dites). Ekspor *raw options*:
```

mazipan commented on src/v2/guide/unit-testing.md:
```diff
@@ -29,36 +30,38 @@ You don't have to do anything special in your components to make them testable.
-// Here are some Jasmine 2.0 tests, though you can
-// use any test runner / assertion library combo you prefer
+// Berikut adalah Jasmine 2.0 tests, Anda dapat
+// menggunakan test runner / library assertion apa pun yang Anda sukai
```
Library -> Pustaka

nusendra commented on src/v2/guide/unit-testing.md:
````diff
@@ -29,36 +30,38 @@ You don't have to do anything special in your components to make them testable.
 ```
 
-Then import the component options along with Vue, and you can make many common assertions (here we are using Jasmine/Jest style `expect` assertions just as an example):
+Kemudian import options komponen tersebut bersama dengan Vue, dan Anda dapat membuat banyak assertion umum (disini kita menggunakan Jasmine/Jest style `expect` assertion sebagai contoh)
````
```suggestion
Kemudian impor opsi dari komponen tersebut bersama dengan Vue, dan Anda dapat membuat banyak *assertion* umum (disini kita menggunakan Jasmine/Jest dengan gaya `expect` *assertion* sebagai contoh)
```

nusendra commented on src/v2/guide/unit-testing.md:
````diff
@@ -29,36 +30,38 @@ You don't have to do anything special in your components to make them testable.
 
 ``` js
-// Import Vue and the component being tested
+// Import vue dan komponen yang akan dites
````
```suggestion
// Impor vue dan komponen yang akan dites
```

nusendra commented on src/v2/guide/unit-testing.md:
```diff
@@ -29,36 +30,38 @@ You don't have to do anything special in your components to make them testable.
 
-// Here are some Jasmine 2.0 tests, though you can
-// use any test runner / assertion library combo you prefer
+// Berikut adalah Jasmine 2.0 tests, Anda dapat
```
```suggestion
// Berikut adalah Jasmine 2.0 tests, Anda bisa
```

mazipan commented on src/v2/guide/unit-testing.md:
````diff
@@ -67,9 +70,9 @@ describe('MyComponent', () => {
 ```
 
-## Writing Testable Components
+## Tulis Komponen yang Dapat Dites
````
Tulis -> Menulis

nusendra commented on src/v2/guide/unit-testing.md:
```diff
@@ -29,36 +30,38 @@ You don't have to do anything special in your components to make them testable.
-// Here are some Jasmine 2.0 tests, though you can
-// use any test runner / assertion library combo you prefer
+// Berikut adalah Jasmine 2.0 tests, Anda dapat
+// menggunakan test runner / library assertion apa pun yang Anda sukai
```
```suggestion
// menggunakan *test runner* / pustaka *assertion* apa pun yang Anda sukai
```

wendyadi replied:
yang ini ga bisa di italic ya karena di dalam comment code

nusendra commented on src/v2/guide/unit-testing.md:
```diff
@@ -29,36 +30,38 @@ You don't have to do anything special in your components to make them testable.
 describe('MyComponent', () => {
-  // Inspect the raw component options
+
+  // Periksa raw options komponen
```
```suggestion
  // Periksa komponen raw options
```

nusendra commented on src/v2/guide/unit-testing.md:
```diff
@@ -29,36 +30,38 @@ You don't have to do anything special in your components to make them testable.
 
-  // Evaluate the results of functions in
-  // the raw component options
+  // Evaluasi hasil dari function-function di
```
```suggestion
  // Evaluasi hasil dari beberapa fungsi di
```

nusendra commented on src/v2/guide/unit-testing.md:
```diff
@@ -29,36 +30,38 @@ You don't have to do anything special in your components to make them testable.
-  // Evaluate the results of functions in
-  // the raw component options
+  // Evaluasi hasil dari function-function di
+  // raw options komponen
```
```suggestion
  // komponen raw options
```

nusendra commented on src/v2/guide/unit-testing.md:
```diff
@@ -67,9 +70,9 @@ describe('MyComponent', () => {
+## Tulis Komponen yang Dapat Dites
 
-A component's render output is primarily determined by the props it receives. If a component's render output solely depends on its props it becomes straightforward to test, similar to asserting the return value of a pure function with different arguments. Take a simplified example:
+Hasil render dari sebuah komponen secara utama ditentukan oleh properti-properti yang diterima komponen tersebut. Jika hasil render dari sebuah komponen hanya bergantung pada properti-nya -- maka akan mudah untuk mengetesnya, mirip dengan melakukan assertion pada fungsi yang memiliki bermacam-macam argumen. Berikut contoh sederhana:
```
```suggestion
Hasil *render* dari sebuah komponen yang paling utama ditentukan oleh properti-properti yang diterima komponen tersebut. Jika hasil *render* dari sebuah komponen hanya bergantung pada properti-nya -- maka hal tersebut akan mudah untuk di tes, mirip dengan menjalankan *assertion* pada fungsi yang memiliki bermacam-macam argumen. Berikut contoh sederhananya:
```
harus menggunakan bahasa yang natural

nusendra commented on src/v2/guide/unit-testing.md:
````diff
@@ -83,13 +86,13 @@ A component's render output is primarily determined by the props it receives. If
 ```
 
-You can assert its render output with different props using the `propsData` option:
+Anda dapat melakukan assertion pada hasil render dengan properti yang berbeda0-beda menggunakan opsi `propsData`:
````
```suggestion
Anda bisa melakukan *assertion* pada hasil *render* menggunakan properti yang berbeda-beda menggunakan opsi `propsData`:
```
bahasa natural dan typo

nusendra commented on src/v2/guide/unit-testing.md:
```diff
@@ -83,13 +86,13 @@ A component's render output is primarily determined by the props it receives. If
 import MyComponent from './MyComponent.vue'
 
-// helper function that mounts and returns the rendered text
+// fungsi helper / pembantu yang melakukan mounts kemudian me-return text yang di-render
```
```suggestion
// fungsi *helper* / pembantu yang menjalankan *mounts* kemudian mengembalikan teks yang di-render
```

nusendra commented on src/v2/guide/unit-testing.md:
```diff
@@ -109,22 +112,23 @@ describe('MyComponent', () => {
+## Menguji/Melakukan Assertion pada Perubahan Asynchronous
 
-Since Vue [performs DOM updates asynchronously](reactivity.html#Async-Update-Queue), assertions on DOM updates resulting from state change will have to be made in a `Vue.nextTick` callback:
+Karena Vue [melakukan update pada DOM secara asynchronous](reactivity.html#Async-Update-Queue), assertion pada update DOM yang disebabkan oleh perubahan state harus dilakukan dalam callback `Vue.nextTick`:
```
```suggestion
Karena Vue [melakukan update pada DOM secara asynchronous](reactivity.html#Async-Update-Queue), *assertion* pada *update* DOM yang disebabkan oleh perubahan *state* harus dilakukan dalam *callback* `Vue.nextTick`:
```

nusendra commented on src/v2/guide/unit-testing.md:
```diff
@@ -109,22 +112,23 @@ describe('MyComponent', () => {
   vm.message = 'foo'
 
-  // wait a "tick" after state change before asserting DOM updates
+  // Tunggu satu tick setelah perubahan state sebelum menguji DOM yang terupdate
```
```suggestion
  // Tunggu satu "tick" setelah perubahan *state* sebelum menguji DOM yang terupdate
```

nusendra commented on src/v2/guide/unit-testing.md:
````diff
@@ -109,22 +112,23 @@ describe('MyComponent', () => {
 ```
 
-For more in-depth information on unit testing in Vue, check out [Vue Test Utils](https://vue-test-utils.vuejs.org/) and our cookbook entry about [unit testing vue components](https://vuejs.org/v2/cookbook/unit-testing-vue-components.html).
+Untuk informasi yang lebih mendalam untuk unit testing di Vue, kunjungi [Vue Test Utils](https://vue-test-utils.vuejs.org/) dan cookboook entry kami tentang [unit testing vue components](https://vuejs.org/v2/cookbook/unit-testing-vue-components.html)
````
```suggestion
Untuk informasi yang lebih mendalam untuk *unit testing* di Vue, kunjungi [Vue Test Utils](https://vue-test-utils.vuejs.org/) dan *cookboook* entri kami tentang [unit testing vue components](https://vuejs.org/v2/cookbook/unit-testing-vue-components.html)
```

nusendra reviewed (changes requested):
Bahasa sudah lumayan natural, cuman masih banyak istilah inggris yang belum di bikin italic.

Mohon bantuannya, terima kasih mas :-)

wendyadi commented:
@mazipan @nusendra 
FYI: Ini di style markdownnya kalau di-italic, warna textnya jadi abu2 ya.
Ini saya italic in dlu

<img width="595" alt="screen shot 2019-03-07 at 22 04 04" src="https://user-images.githubusercontent.com/13862336/53966067-0087b580-4125-11e9-9ead-fe7e76fdc28e.png">

wendyadi commented:
@nusendra @mazipan 
Another ss yak:

<img width="621" alt="screen shot 2019-03-07 at 22 13 35" src="https://user-images.githubusercontent.com/13862336/53966711-46914900-4126-11e9-9b29-3d0c0b04165f.png">

wendyadi committed:
memberi italic pada beberapa istilah dan memperbaiki pemilihan kata

mazipan commented:
Periksa ulang komentar2nya ya mas @wendyadi

wendyadi committed:
fix line 95

mazipan commented on src/v2/guide/unit-testing.md:
```diff
@@ -67,9 +70,9 @@ describe('MyComponent', () => {
+## Menulis Komponen yang Dapat Dites
 
-A component's render output is primarily determined by the props it receives. If a component's render output solely depends on its props it becomes straightforward to test, similar to asserting the return value of a pure function with different arguments. Take a simplified example:
+Hasil *render* dari sebuah komponen secara utama ditentukan oleh properti-properti yang diterima komponen tersebut. Jika hasil *render* dari sebuah komponen hanya bergantung pada properti-nya -- maka akan mudah untuk mengetesnya, mirip dengan melakukan assertion pada fungsi yang memiliki bermacam-macam argumen. Berikut contoh sederhana:
```
props disini kan bukan properti.

mazipan replied:
props tetep pakai props

mazipan commented on src/v2/guide/unit-testing.md:
````diff
@@ -109,22 +112,23 @@ describe('MyComponent', () => {
 ```
 
-For more in-depth information on unit testing in Vue, check out [Vue Test Utils](https://vue-test-utils.vuejs.org/) and our cookbook entry about [unit testing vue components](https://vuejs.org/v2/cookbook/unit-testing-vue-components.html).
+Untuk informasi yang lebih mendalam untuk *unit testing* di Vue, kunjungi [Vue Test Utils](https://vue-test-utils.vuejs.org/) dan *cookboook entry* kami tentang [unit testing vue components](https://vuejs.org/v2/cookbook/unit-testing-vue-components.html)
````
cookbook entry -> buku panduan

mazipan commented on src/v2/guide/unit-testing.md:
```diff
@@ -109,22 +112,23 @@ describe('MyComponent', () => {
+## Melakukan Pengujian/*Assertion* pada Perubahan Asynchronous
 
-Since Vue [performs DOM updates asynchronously](reactivity.html#Async-Update-Queue), assertions on DOM updates resulting from state change will have to be made in a `Vue.nextTick` callback:
+Karena Vue [melakukan update pada DOM secara asynchronous](reactivity.html#Async-Update-Queue), *assertion* pada *update* DOM yang disebabkan oleh perubahan *state* harus dilakukan dalam callback `Vue.nextTick`:
```
update -> pembaruan

wendyadi committed:
request change:
 - props tetep pakai props
 - cookbook entry -> buku panduan
 - update -> pembaruan
